### PR TITLE
Add list deployments method to admin router

### PR DESCRIPTION
**Pull request by larioj**
This is necessary to check that deployments have finished.

### CODE REVIEW

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala [hunk 1]
```diff
@@ -81,4 +86,16 @@ object ItUtil {
     )
   }
 
+  def waitForDeployment(adminRouter: AdminRouter)(attempts: Int): Boolean = {
+    Stream.tabulate(attempts) { _ =>
+      Thread.sleep(1.second.toMillis)
+      val response = Await.result {
+        adminRouter.listDeployments()
+      }
+      assert(response.status == Status.Ok || response.status == Status.Conflict)
```
Comment (cruhland) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala]: I don't think the list deployments endpoint returns a 409 status?

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala [hunk 2]
```diff
@@ -81,4 +86,16 @@ object ItUtil {
     )
   }
 
+  def waitForDeployment(adminRouter: AdminRouter)(attempts: Int): Boolean = {
+    Stream.tabulate(attempts) { _ =>
+      Thread.sleep(1.second.toMillis)
```
Comment (jsancio) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala]: We should be able to do this async. I think you want something similar to what we do here: https://github.com/dcos/cosmos/blob/master/cosmos-test-common/src/main/scala/com/mesosphere/cosmos/test/TestUtil.scala#L50

Twitter Future don't use more memory when the Future recursion is tail recursive.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala (hunk 1 above)
Comment (larioj) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala]: You are correct. This should not be here.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala (hunk 2 above)
Comment (larioj) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala]: This does not timeout?

Comment (larioj) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/ItUtil.scala]: This is just for integration tests. We are going to await it anyways.

**Pull request merged**